doc/users/next_whats_new/boxplot_legend_support.rst

```diff
@@ -0,0 +1,37 @@
+Legend support for Boxplot
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+Boxplots now support a *label* parameter to create legend entries.
+Legend labels can be passed as a string to each individual boxplot
+or as a list of strings to label multiple boxes in a single
+boxplot call.
+
+
+.. plot::
```

Comment:
I suggest to also add a plot for the single-string case. Something like https://github.com/matplotlib/matplotlib/pull/27840#discussion_r1518807844 - you may switch to `patch_artist=True` and color the boxes so that the distinction between the two sets is more obvious than just the color of the medians.

Reply:
Do we want this behaviour when boxplots with patches are mixed with ones without?

![2box](https://github.com/matplotlib/matplotlib/assets/9763064/91651909-68d0-4899-933b-1b10ab059403)

Reply:
There’s no way around this. Tge individual boxplots do not know about each other and a non-patch_artist, boxplot can only have a line as legend entry, whereas the patch_artist boxplots should have the box. It’s the responsibility of the user to create a consistent plot.

Reply:
Just wondering: Shouldn’t the color of the line for data A not be the orange of the median line?

Reply:
Yes it should! For some reason it only does if showbox=False. I didn't notice it before